### DISCUSSION TITLE
Randomness in Determinism

### DISCUSSION THREAD

**Discussion by michael-camilleri** [Project-MONAI/MONAI#8254] [Q&A]
Hi

This is just to check that my understanding is correct. When using set_determinism(), together with the appropriate monai data_loader, this just ensures that the random number generation is based on a seed and hence reproducible, but is otherwise still 'pseudo-random'. Am I right?

Thanks

**Comment by KumoLiu** [comment]
Hi @michael-camilleri, thanks for your interest here.

I’m not sure I fully understand your point here. If you’re using a random transform, it should indeed be performed randomly. The `set_determinism` function in MONAI is primarily designed to ensure reproducibility in your experiments. As we all know, deep learning experiments can be challenging to reproduce. Once you achieve a good result, having a reproducible setup allows you to apply the same configuration effectively in subsequent downstream tasks.

Hope it helps, thanks.
